src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -736,6 +758,40 @@ class Config:
 
     @accepts()
     async def providers(self):
+        """
+        Returns a list of dictionaries of supported providers for Cloud Sync Tasks.
+
+        Example of a single provider:
```

Comment:
`credentials_schema` - JSON Schema for credentials attributes
`task_schema` - JSON Schema for task attributes
`buckets` - true if provider uses buckets